homeassistant/components/honeywell/climate.py

```diff
@@ -126,31 +126,39 @@ def __init__(self, client, device, cool_away_temp,
         self._username = username
         self._password = password
 
-        self._supported_features = (SUPPORT_PRESET_MODE |
-                                    SUPPORT_TARGET_TEMPERATURE_RANGE)
-
         # pylint: disable=protected-access
-        _LOGGER.debug("uiData = %s ", device._data['uiData'])
+        _LOGGER.debug("latestData = %s ", device._data)
 
         # not all honeywell HVACs upport all modes
         mappings = [v for k, v in HVAC_MODE_TO_HW_MODE.items()
                     if k in device._data['uiData']]
         self._hvac_mode_map = {k: v for d in mappings for k, v in d.items()}
 
+        self._supported_features = SUPPORT_PRESET_MODE
+
+        if device._data['uiData']['SwitchAutoAllowed'] or \
+                (device._data['uiData']['SwitchCoolAllowed'] and
+                 device._data['uiData']['SwitchHeatAllowed']):
+            self._supported_features |= SUPPORT_TARGET_TEMPERATURE_RANGE
+        else:
+            self._supported_features |= SUPPORT_TARGET_TEMPERATURE
```

Comment:
I am not sure to have either:
 - one of `SUPPORT_TARGET_TEMPERATURE_RANGE`, _or_ `SUPPORT_TARGET_TEMPERATURE`, or simply 
 - `SUPPORT_TARGET_TEMPERATURE` in all circumstances

Reply:
Set both for now. Then just return None from min/max when range is not supported.

Reply:
Done